sql: limit "canceling all jobs due to liveness failure'

tbg opened the pull request:
Seen about every second in #28179.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/28518)
<!-- Reviewable:end -->

a-robinson reviewed (approved):
LGTM I suppose, but it seems like it'd be much better to conditionally log if and only if `r.mu.jobs` is non-empty.

tbg committed:
sql: limit "canceling all jobs due to liveness failure'

Seen about every second in #28179. It turns out that the subsequent call
to `cancelAll` already logs a message for each job (which is then
removed), which is sufficient.

Release note: None

tschottdorf commented:
Good call, @a-robinson. Looking into this, I realized that `cancelAll` was already logging a message for each job. So I just deleted the spammy log line.

tschottdorf commented:
bors r=a-robinson
